File: atcoder/atcoder_problems/abc/abc101~200/abc101~150/abc122/abc122_b.py
S = input()


T = ["A", "C", "G", "T"]
ans = 0
N = len(S)

for i in range(N):
    for j in range(i, N):
        ok = True
        for k in range(i, j+1):
            if S[k] not in T:
                ok = False
        if ok:
            ans = max(ans, j - i + 1)

print(ans)



# S = input()

# ans = 0
# temp = 0
# for s in S:
#     if s == "A" or s == "C" or s == "G" or s == "T":
#         temp += 1
#     else:
#         ans = max(ans, temp)
#         temp = 0

# ans = max(ans, temp)
# print(ans)

File: atcoder/atcoder_problems/abc101~200/abc101~150/abc122/abc122_b.cpp
#include <bits/stdc++.h>
using namespace std;
#include <atcoder/all>
using namespace atcoder;
#define rep(i,n) for (int i = 0; i < (n); ++i)

int main() {
    string S;
    cin >> S;

    string T = "ATCG";

    int ans = 0;
    int now = 0;
    for (int i = 0; i < S.size(); i++) {
        bool isATCG = false;
        for (int j = 0; j < T.size(); j++){
            if (S[i] == T[j]) isATCG = true;
        }

        if (!isATCG) {
            now = 0;
        }
        else {
            now++;
            ans = max(ans, now);
        }
    }
    cout << ans << endl;
    return 0;
}



// #include <bits/stdc++.h>
// using namespace std;
// #include <atcoder/all>
// using namespace atcoder;
// #define rep(i,n) for (int i = 0; i < (n); ++i)


// int main() {
//     string S;
//     cin >> S;
//     int ans = 0;
//     int cnt = 0;
//     for (char s : S) {
//         if (s == 'A' or s == 'C' or s == 'G' or s == 'T') cnt++;
//         else {
//             ans = max(ans, cnt);
//             cnt = 0;
//         }
//     }
//     ans = max(ans, cnt);
//     cout << ans << endl;
//     return 0;
// }